ClassHierarchyAndCollections/ClassHierarchyAndCollectionsTests/AssociationTests.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using ClassHierarchyAndCollections;
+
+namespace ClassHierarchyAndCollectionsTests
+{
+    /// <summary>
+    /// Defines tests for the <see cref="Association"/> class.
+    /// </summary>
```

Comment:
I really like that you documented your test classes and methods as well. You don't have to do this going forward, but I think it is helpful to other developers when you do that (+0.5)